pkg/cluster/mpi/statusupdate.go

```diff
@@ -23,8 +23,15 @@ const (
 	StartingStatus v1.JobConditionType = "Starting"
 	RunningStatus  v1.JobConditionType = "Running"
 	StoppingStatus v1.JobConditionType = "Stopping"
+	FailedStatus   v1.JobConditionType = "Failed"
 )
 
+// failureReasons lists generic Pod failure reasons that cause FailedStatus of MPI Cluster,
+// and map them to more specific strings that are reported in MPI CRD.
+var failureReasons = map[string]string{
+	"CrashLoopBackOff": "EntryPointFailed",
```

Comment:
I'm not sure what all could lead to this condition, can we check like the exit code on the container, or something more specific?

Reply:
It's possible to obtain container's last exit code, but it doesn't look more informative than `CrashLoopBackOff`. The only advantage I see is that the we know what an exit code is, while exact meanings of Kubernetes events are not well documented. I have another idea, though: In the current design we do not handle legitimate container restarts. That is, MPI's Failed status must only be used if the container has never managed to run. I will work on that.